src/__internal__/sticky-footer/sticky-footer.component.js

```diff
@@ -0,0 +1,65 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import PropTypes from "prop-types";
+import throttle from "lodash/throttle";
+
+import StyledStickyFooter from "./sticky-footer.style";
+
+const SCROLL_THROTTLE = 50;
+
+const StickyFooter = ({ children, containerRef, disableSticky, ...rest }) => {
+  const [isSticky, setIsSticky] = useState(true);
+  const footerRef = useRef();
+
+  const checkFooterPosition = useCallback(
+    throttle(() => {
+      let fullyScrolled;
+      const content = containerRef.current;
+      const stickyOffset = footerRef.current.clientHeight / 2;
+
+      if (
+        content.scrollHeight - content.scrollTop - stickyOffset <=
+        content.clientHeight
+      ) {
+        fullyScrolled = true;
+      }
+
+      if (fullyScrolled) {
+        setIsSticky(false);
+      } else {
+        setIsSticky(true);
+      }
+    }, SCROLL_THROTTLE),
+    []
+  );
+
+  useEffect(() => {
+    const content = containerRef.current;
+    content.addEventListener("scroll", checkFooterPosition, false);
+
+    return () => {
+      content.removeEventListener("scroll", checkFooterPosition, false);
+    };
+  }, [checkFooterPosition, containerRef]);
+
+  return (
+    <StyledStickyFooter
+      data-component="sticky-footer"
+      sticky={disableSticky ? false : isSticky}
+      ref={footerRef}
+      {...rest}
+    >
+      {children}
+    </StyledStickyFooter>
+  );
+};
+
+StickyFooter.propTypes = {
+  /** child elements */
+  children: PropTypes.node.isRequired,
+  /** Ref of the container the footer should be sticky on */
+  containerRef: PropTypes.any.isRequired,
+  /** Disable the sticky behaviour manually */
+  disableSticky: PropTypes.bool,
```

Comment:
If it's not a required prop, do we need to have a default prop for it?

Reply:
No we don't, this prop will only affect the component if it's set to `true`

Reply:
I am used to the fact that if a prop is not required it must have a default value (eslint rule). But that's ok for me. Thanks

Reply:
Yeah we have the `"react/require-default-props"` eslint rule turned off at the moment, we have `undefined` props all over the place in Carbon!

Maybe something we could think about changing in the future though